### ISSUE TITLE
NoItParamInMultilineLambdaRule doesn't account for variables or functions that could be named "it"

### ISSUE THREAD

**Issue by Trixmegist** [pinterest/ktlint#306]
NoItParamInMultilineLambdaRule simply checks that there're no identifiers named "it" inside a multiline lambda. That makes it impossible to use this rule with several testing frameworks, e.g. Kotlintest, because they require to use "it()" functions inside the specs:

```
class MySpec : DescribeSpec({
    describe("if this") {
        it("does that") { // NoItParamInMultilineLambdaRule doesn't allow "it"
        }
    })
```

You already have a "fixme" in the source code of the rule, so I'm wondering when is this going to be fixed?
```
class NoItParamInMultilineLambdaRule : Rule("no-it-in-multiline-lambda") {

    override fun visit(...) {
        // fixme: we are not accounting for "it" variable that can be defined in the same scope
        ...
    }
}
```

**Comment by ofdrykkja** [comment]
from which code-style is this rule? can't find this suggestion anywhere expect ktlint

**Comment by Trixmegist** [comment]
> from which code-style is this rule? can't find this one anywhere expect ktlint

https://github.com/shyiko/ktlint/blob/master/ktlint-ruleset-standard/src/main/kotlin/com/github/shyiko/ktlint/ruleset/standard/NoItParamInMultilineLambdaRule.kt

**Comment by ofdrykkja** [comment]
i mean what document is this rule based on? i asked because didn't find nothing similar nor in android code style guide nor in kotlin code style guide

**Comment by ofdrykkja** [comment]
just can't understand for whats reason i must specify explicitly name of "it" in multiline lambda case

**Comment by shyiko** [comment]
@ofdrykkja https://github.com/yole/kotlin-style-guide/issues/10#issue-157736558

**Comment by sowmyav24** [comment]
The `// fixme: we are not accounting for "it" variable that can be defined in the same scope` has been addressed in this [PR #507](https://github.com/pinterest/ktlint/pull/507) along with accounting `it` as a function call for frameworks such as `Kotlintest`

In `StandardRuleSetProvider`, `NoItParamInMultilineLambdaRule` is disabled for reason `// disabled until it's clear what to do in case of 'import _.it' `. Allowing function call would solve this issue. Is there anything else that is missing for this rule to be enabled ?

**Comment by sowmyav24** [comment]
We can close this issue as the rule is no longer supported
